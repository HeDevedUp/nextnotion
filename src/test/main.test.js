import { jest } from 'jest';


import { createTag } from '../';

describe('createTag', () => {
  let input;

  beforeEach(() => {
    input = document.createElement('input');
    input.textContent = '  Test Input';
  });

  test('it adds class "input" and tag type class', () => {
    createTag(input, 'h1');
    expect(input.classList.contains('input')).toBe(true);
    expect(input.classList.contains('h1')).toBe(true);
  });

  test('it removes the first 2 characters from text content and trims the rest', () => {
    createTag(input, 'h2');
    expect(input.textContent).toBe('Test Input');
  });

  test('it sets placeholder for heading tag types', () => {
    createTag(input, 'h3');
    expect(input.getAttribute('placeholder')).toBe('Heading h3');
  });

  test('it sets placeholder for paragraph tag type', () => {
    createTag(input, 'p');
    expect(input.getAttribute('placeholder')).toBe('Paragraph');
  });

  test('it focuses on the input', () => {
    jest.spyOn(input, 'focus');
    createTag(input, 'p');
    expect(input.focus).toHaveBeenCalled();
  });
});
