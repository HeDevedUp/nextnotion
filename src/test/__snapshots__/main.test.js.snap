// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup functionality creates an option element with the correct structure and data 1`] = `
"
    <button class=\\"popup__item\\" data-type=\\"h1\\">
    <img src=\\"\\" alt=\\"heading icon\\" width=\\"30\\">

    <div>
      <h4>Heading 1</h4>
      <p>Big Section heading</p>
    </div>
    </button>"
`;
